### ISSUE TITLE
COMPOSE_CONVERT_WINDOWS_PATHS not working on v2

### ISSUE THREAD

**Issue by blerins** [docker/compose#9132]
Hello, I am having trouble to use docker-compose v2 on Windows without Docker Desktop with volumes. I have Docker installed on WSL2 (Ubuntu distro) with remote API enabled.

On Windows I am using Docker CLI with the DOCKER_HOST variable setted up.

Environment:
```
PS C:\Users\bruno.lerin\workspace\dc> cat .\docker-compose.yml
version: '3.6'

services:
  test:
    image: hello-world
    volumes:
      - ./:/test
PS C:\Users\bruno.lerin\workspace\dc> echo $Env:COMPOSE_CONVERT_WINDOWS_PATHS
1
PS C:\Users\bruno.lerin\workspace\dc> docker --version
Docker version 20.10.12, build e91ed57
PS C:\Users\bruno.lerin\workspace\dc> docker-compose config
services:
  test:
    image: hello-world
    volumes:
    - /c/Users/bruno.lerin/workspace/dc:/test:rw
version: '3.6'

PS C:\Users\bruno.lerin\workspace\dc> docker-compose2 config
services:
  test:
    image: hello-world
    networks:
      default: null
    volumes:
    - type: bind
      source: C:\Users\bruno.lerin\workspace\dc
      target: /test
      bind:
        create_host_path: true
networks:
  default:
    name: dc_default
```

Using docker-compose up on v2 is failing
```
PS C:\Users\bruno.lerin\workspace\dc> docker-compose2 -v
Docker Compose version v2.2.3
PS C:\Users\bruno.lerin\workspace\dc> docker-compose2 up
[+] Running 0/0
 - Container 4c1d77642ad3_dc_test_1  Recreate                                                                                                                                                        0.0s
Error response from daemon: invalid volume specification: 'C:\Users\bruno.lerin\workspace\dc:/test:rw'
```

While doing it so on v1 succeds
```
PS C:\Users\bruno.lerin\workspace\dc> docker-compose -v
docker-compose version 1.29.2, build 5becea4c
PS C:\Users\bruno.lerin\workspace\dc> docker-compose up
Starting 4c1d77642ad3_4c1d77642ad3_dc_test_1 ... done
Attaching to 4c1d77642ad3_4c1d77642ad3_dc_test_1
4c1d77642ad3_4c1d77642ad3_dc_test_1 |
4c1d77642ad3_4c1d77642ad3_dc_test_1 | Hello from Docker!
4c1d77642ad3_4c1d77642ad3_dc_test_1 | This message shows that your installation appears to be working correctly.
[...]
4c1d77642ad3_4c1d77642ad3_dc_test_1 |
4c1d77642ad3_4c1d77642ad3_dc_test_1 exited with code 0
```

In the config command we already see the issue with the mapped issue. I have been unable to find an altertanive for COMPOSE_CONVERT_WINDOWS_PATHS on v2.

If you need any further information, please let me know.

Kind regards.

**Comment by quequiere** [comment]
I have the same problem here, COMPOSE_CONVERT_WINDOWS_PATHS doesn't change anything.

**Comment by VedranVukoje** [comment]
Hi. I got same issue. I am using docker with minikube . [Docker desktop replacement](https://minikube.sigs.k8s.io/docs/tutorials/docker_desktop_replacement/)
I also try with docker-machine I got same error.
`Error response from daemon: invalid volume specification: 'C:\Users`

$ docker compose version
Docker Compose version v2.2.3

Kind regards.

**Comment by ndeloof** [comment]
I wonder why we need this; Doesn't docker cli support passing `-v c:\foo:/foo` to bind a local windows path inside container?  Then docker CLI code has the required logic and docker/compose should use the same

**Comment by kohanyirobert** [comment]
I'm confused a bit. I'm using `docker-compose` 2.4.1 and as I've seen [this version uses `compose-go` 1.2.2](https://github.com/docker/compose/blob/v2.4.1/go.mod#L9) and [which contains a fix related to this problem](https://github.com/compose-spec/compose-go/releases/tag/v1.2.2).

With a `docker-compose.yaml` containing a bind mount with a host path of `.` or `./` when I execute `docker-compose up` I receive the following error.

```ps1
> docker-compose up
[+] Running 3/4
 - Network app_default           Created                                                                                        0.1s
 - Volume "app_todo-mysql-data"  Created                                                                                        0.0s
 - Container app-mysql-1         Created                                                                                        0.1s
 - Container app-app-1           Creating                                                                                       0.1s
Error response from daemon: invalid volume specification: 'C:\Users\rkohanyi\Downloads\app:/app:rw'
```

If I replace the relative reference to the current working directory with an absolute Windows path to it like `c:\Users\rkohanyi\Downloads\app:/app` or `c:/Users/rkohanyi/Downloads/app:/app` things works as expected.

**Shouldn't relative references be converted when they evaluate to a Windows path when this environment variable is set?**
Not sure, but this would be my expectation (as I have seen the fix included in 1.2.2 of `compose-go` didn't test for this edge case.
As a workaround I use `.env` and I set `PWD=c:\Users\rkohanyi\Downloads\app` which works (but then one could be using this solution to workaround _all_ problems related to Windows paths).

Versions and environment variable info.

```ps1
> docker-compose version
Docker Compose version v2.4.1
> echo $env:COMPOSE_CONVERT_WINDOWS_PATHS
1
```

**YAML with relative path.**

```ps1
> cat .\docker-compose.yaml
services:
  app:
    image: node:12-alpine
    command: sh -c "yarn install && yarn run dev"
    working_dir: /app
    ports:
      - 3000:3000
    volumes:
      - ./:/app
    environment:
      MYSQL_HOST: mysql
      MYSQL_USER: root
      MYSQL_PASSWORD: secret
      MYSQL_DB: todos
  mysql:
    image: mysql:5.7
    volumes:
      - todo-mysql-data:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD: secret
      MYSQL_DATABASE: todos

volumes:
  todo-mysql-data:
> docker-compose config
name: app
services:
  app:
    command:
    - sh
    - -c
    - yarn install && yarn run dev
    environment:
      MYSQL_DB: todos
      MYSQL_HOST: mysql
      MYSQL_PASSWORD: secret
      MYSQL_USER: root
    image: node:12-alpine
    networks:
      default: null
    ports:
    - mode: ingress
      target: 3000
      published: "3000"
      protocol: tcp
    volumes:
    - type: bind
      source: ./
      target: /app
      bind:
        create_host_path: true
    working_dir: /app
  mysql:
    environment:
      MYSQL_DATABASE: todos
      MYSQL_ROOT_PASSWORD: secret
    image: mysql:5.7
    networks:
      default: null
    volumes:
    - type: volume
      source: todo-mysql-data
      target: /var/lib/mysql
      volume: {}
networks:
  default:
    name: app_default
volumes:
  todo-mysql-data:
    name: app_todo-mysql-data
```

**YAML with absolute path.**

```ps1
> cat .\docker-compose.yaml
services:
  app:
    image: node:12-alpine
    command: sh -c "yarn install && yarn run dev"
    working_dir: /app
    ports:
      - 3000:3000
    volumes:
      - c:\Users\rkohanyi\Downloads\app:/app
    environment:
      MYSQL_HOST: mysql
      MYSQL_USER: root
      MYSQL_PASSWORD: secret
      MYSQL_DB: todos
  mysql:
    image: mysql:5.7
    volumes:
      - todo-mysql-data:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD: secret
      MYSQL_DATABASE: todos

volumes:
  todo-mysql-data:
> docker-compose config
name: app
services:
  app:
    command:
    - sh
    - -c
    - yarn install && yarn run dev
    environment:
      MYSQL_DB: todos
      MYSQL_HOST: mysql
      MYSQL_PASSWORD: secret
      MYSQL_USER: root
    image: node:12-alpine
    networks:
      default: null
    ports:
    - mode: ingress
      target: 3000
      published: "3000"
      protocol: tcp
    volumes:
    - type: bind
      source: /c/Users/rkohanyi/Downloads/app
      target: /app
      bind:
        create_host_path: true
    working_dir: /app
  mysql:
    environment:
      MYSQL_DATABASE: todos
      MYSQL_ROOT_PASSWORD: secret
    image: mysql:5.7
    networks:
      default: null
    volumes:
    - type: volume
      source: todo-mysql-data
      target: /var/lib/mysql
      volume: {}
networks:
  default:
    name: app_default
volumes:
  todo-mysql-data:
    name: app_todo-mysql-data
```
